service/src/test/java/bio/terra/workspace/service/privateresource/PrivateResourceCleanupServiceTest.java

```diff
@@ -0,0 +1,337 @@
+    // Verify second user is no longer in workspace, but still has resource access because cleanup
+    // hasn't run yet.
+    assertFalse(
+        SamRethrow.onInterrupted(
+            () ->
+                samService.isAuthorized(
+                    userAccessUtils.secondUserAuthRequest(),
+                    SamResource.WORKSPACE,
+                    resource.getWorkspaceId().toString(),
+                    SamWorkspaceAction.READ),
+            "checkResourceAuth"));
+    assertTrue(
+        SamRethrow.onInterrupted(
+            () ->
+                samService.isAuthorized(
+                    userAccessUtils.secondUserAuthRequest(),
+                    resource.getCategory().getSamResourceName(),
+                    resource.getResourceId().toString(),
+                    SamControlledResourceActions.READ_ACTION),
+            "checkResourceAuth"));
+    // Manually enable and run cleanup.
+    privateResourceCleanupConfiguration.setEnabled(true);
+    // Calling "cleanupResources" manually lets us skip waiting for the cronjob to trigger.
+    privateResourceCleanupService.cleanupResourcesSuppressExceptions();
+    // Verify second user can no longer read the resource.
+    assertFalse(
+        SamRethrow.onInterrupted(
+            () ->
+                samService.isAuthorized(
+                    userAccessUtils.secondUserAuthRequest(),
+                    resource.getCategory().getSamResourceName(),
+                    resource.getResourceId().toString(),
+                    SamControlledResourceActions.READ_ACTION),
+            "checkResourceAuth"));
+    // Verify resource is marked "abandoned"
+    ControlledResource dbResource =
+        resourceDao
+            .getResource(resource.getWorkspaceId(), resource.getResourceId())
+            .castToControlledResource();
+    assertEquals(PrivateResourceState.ABANDONED, dbResource.getPrivateResourceState().get());
+    // Application can still read the resource, because applications have EDITOR role on their
+    // application-private resources.
+    assertTrue(
+        SamRethrow.onInterrupted(
+            () ->
+                samService.isAuthorized(
+                    appRequest,
+                    resource.getCategory().getSamResourceName(),
+                    resource.getResourceId().toString(),
+                    SamControlledResourceActions.READ_ACTION),
+            "checkResourceAuth"));
+  }
+
+  private App getAppBySa(String serviceAccount) {
+    for (App app : appConfig.getConfigurations()) {
+      if (app.getServiceAccount().equals(serviceAccount)) {
+        return app;
+      }
+    }
+    throw new NotFoundException("No application configured with service account " + serviceAccount);
+  }
+
+  /**
+   * Create a Sam managed group and return its email. This functionality is only used by tests, so
+   * it lives here instead of in SamService.
+   */
+  private String createGroup(String groupName, GroupApi groupApi) {
```

Comment:
static. i guess there's no other usage yet to be worth having a separate util?

Reply:
Pretty much, yep. I'd pull this into SamService if we need it for server code or into a static util class if other tests needed it, but for now this is the only usage.